src/control/server/ctl_harness.go

```diff
@@ -31,78 +31,145 @@ import (
 
 	"github.com/daos-stack/daos/src/control/common/proto/convert"
 	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/daos-stack/daos/src/control/logging"
 	"github.com/daos-stack/daos/src/control/system"
 )
 
-type HarnessAction int
+// defaultTimeout used in harness requests.
+const defaultTimeout = 10 * time.Second
 
-const (
-	HarnessUnknown HarnessAction = iota
-	HarnessQuery
-	HarnessPrepShutdown
-	HarnessStop
-	HarnessStart
-
-	defaultTimeout = 10 * time.Second
-)
-
-func (hs HarnessAction) String() string {
-	return [...]string{
-		"Unknown",
-		"Query",
-		"PrepShutdown",
-		"Stop",
-		"Start",
-	}[hs]
+// HarnessClient provides methods for requesting actions on remote harnesses
+// (attached to listening gRPC servers) identified by host addresses.
+//
+// Results will be provided for each rank (member) managed by harness.
+// Single gRPC sent over management network per harness (not per rank).
+type HarnessClient interface {
+	Query(context.Context, string, ...uint32) (system.MemberResults, error)
+	PrepShutdown(context.Context, string, ...uint32) (system.MemberResults, error)
+	Stop(context.Context, string, bool, ...uint32) (system.MemberResults, error)
+	Start(context.Context, string, ...uint32) (system.MemberResults, error)
```

Comment:
@kjacque asked for an interface so I provided one, is there a problem with having one? it will be possible to decouple and provide more detailed tests in individual layers rather than providing lots of test cases for the bigger chunk of logic that would be tested with mock mgmtSvcClient. if a change is requested I would prefer to make it in a follow-up PR where I will be adding tests for ctl_system so we can get this PR landed.

Reply:
I could go either way as long as we keep these friendlier public method names (vs. calling harnessAction with lots of params). I do think if you plan to add tests for users of the HarnessClient methods, an interface makes it a lot easier to test that layer.